src/main/java/de/tum/in/www1/artemis/repository/TeamRepository.java

```diff
@@ -98,7 +107,8 @@ default Team save(Exercise exercise, Team team) {
             team.setLastModifiedDate(Instant.now());
         }
         team.setExercise(exercise);
-        return save(team);
+        team = save(team);
+        return findWithStudentsByIdElseThrow(team.getId());
```

Comment:
do we have to re-fetch, or can't we just manually set the already loaded students?